zerver/lib/streams.py

```diff
@@ -521,6 +521,17 @@ def access_stream_for_delete_or_update_requiring_metadata_access(
     return (stream, sub)
 
 
+def has_metadata_access_to_channel_via_groups(
+    user_recursive_group_ids: set[int],
+    can_administer_channel_group_id: int,
+    can_add_subscribers_group_id: int,
```

Comment:
It's not amazing that the order of these two fields is not obvious when reading a caller, but given they're used identically, it's probably OK. I think we can think about whether there's a nicer interface for this function once the functionality is merged.